# Projeto Roof Imóveis

In [106]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [107]:
df = pd.read_csv('kc_house_data.csv')
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [108]:
# Em primeiro lugar, tratei de identificar se havia algum tipo de dado que tivesse que ser mudado.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [109]:
# Notei que era necessário mudar a data para datetime, os banheiros para inteiros, os andares para inteiros e a condição do imóvel ser beira-mar para boolean.
df['date'] = pd.to_datetime(df['date'])
df['bathrooms'] = df['bathrooms'].astype('int64')
df['floors'] = df['floors'].astype('int64')
df['waterfront'] = df['waterfront'].astype('boolean')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   id             21613 non-null  int64         
 1   date           21613 non-null  datetime64[ns]
 2   price          21613 non-null  float64       
 3   bedrooms       21613 non-null  int64         
 4   bathrooms      21613 non-null  int64         
 5   sqft_living    21613 non-null  int64         
 6   sqft_lot       21613 non-null  int64         
 7   floors         21613 non-null  int64         
 8   waterfront     21613 non-null  boolean       
 9   view           21613 non-null  int64         
 10  condition      21613 non-null  int64         
 11  grade          21613 non-null  int64         
 12  sqft_above     21613 non-null  int64         
 13  sqft_basement  21613 non-null  int64         
 14  yr_built       21613 non-null  int64         
 15  yr_renovated   2161

In [110]:
# Em seguida, tratei de eliminar qualquer dado nulo que pudesse estar no meu dataset, mas pelo que vi, não havia nenhum dado nulo.
df = df.dropna()
df

id       date     price  bedrooms  bathrooms  sqft_living  \
0      7129300520 2014-10-13  221900.0         3          1         1180   
1      6414100192 2014-12-09  538000.0         3          2         2570   
2      5631500400 2015-02-25  180000.0         2          1          770   
3      2487200875 2014-12-09  604000.0         4          3         1960   
4      1954400510 2015-02-18  510000.0         3          2         1680   
...           ...        ...       ...       ...        ...          ...   
21608   263000018 2014-05-21  360000.0         3          2         1530   
21609  6600060120 2015-02-23  400000.0         4          2         2310   
21610  1523300141 2014-06-23  402101.0         2          0         1020   
21611   291310100 2015-01-16  400000.0         3          2         1600   
21612  1523300157 2014-10-15  325000.0         2          0         1020   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
0          5650       1       False     0  ...      7        1180   
1          7242       2       False     0  ...      7        2170   
2         10000       1       False     0  ...      6         770   
3          5000       1       False     0  ...      7        1050   
4          8080       1       False     0  ...      8        1680   
...         ...     ...         ...   ...  ...    ...         ...   
21608      1131       3       False     0  ...      8        1530   
21609      5813       2       False     0  ...      8        2310   
21610      1350       2       False     0  ...      7        1020   
21611      2388       2       False     0  ...      8        1600   
21612      1076       2       False     0  ...      7        1020   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0                  0      1955             0    98178  47.5112 -122.257   
1                400      1951          1991    98125  47.7210 -122.319   
2                  0      1933             0    98028  47.7379 -122.233   
3                910      1965             0    98136  47.5208 -122.393   
4                  0      1987             0    98074  47.6168 -122.045   
...              ...       ...           ...      ...      ...      ...   
21608              0      2009             0    98103  47.6993 -122.346   
21609              0      2014             0    98146  47.5107 -122.362   
21610              0      2009             0    98144  47.5944 -122.299   
21611              0      2004             0    98027  47.5345 -122.069   
21612              0      2008             0    98144  47.5941 -122.299   

       sqft_living15  sqft_lot15  
0               1340        5650  
1               1690        7639  
2               2720        8062  
3               1360        5000  
4               1800        7503  
...              ...         ...  
21608           1530        1509  
21609           1830        7200  
21610           1020        2007  
21611           1410        1287  
21612           1020        1357  

[21613 rows x 21 columns]

In [111]:
# Após a eliminação dos dados nulos, criei uma nova coluna para identificar o preço por área de cada imóvel.
df['price_per_area'] = df['price'] / df['sqft_lot']
df.head()

id       date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520 2014-10-13  221900.0         3          1         1180   
1  6414100192 2014-12-09  538000.0         3          2         2570   
2  5631500400 2015-02-25  180000.0         2          1          770   
3  2487200875 2014-12-09  604000.0         4          3         1960   
4  1954400510 2015-02-18  510000.0         3          2         1680   

   sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
0      5650       1       False     0  ...        1180              0   
1      7242       2       False     0  ...        2170            400   
2     10000       1       False     0  ...         770              0   
3      5000       1       False     0  ...        1050            910   
4      8080       1       False     0  ...        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  price_per_area  
0        5650       39.274336  
1        7639       74.288870  
2        8062       18.000000  
3        5000      120.800000  
4        7503       63.118812  

[5 rows x 22 columns]

In [113]:
# Chegando na parte de análise, tratei de identificar os imóveis a beira-mar, com grau 4 de beleza da vista e com preços menores que $500.000,00.
# Além disso, ordenei os imóveis por ordem decrescente de área do terreno, dessa forma identifiquei 5 imóveis baratos com grande possibilidade de valorização após construir no terreno.
# Apenas dois me chamaram atenção: id = 2013802030 e id = 2013802060, os dois vendidos a um preço baixo, mas com grande área de terreno, preço por area de aproximadamente 6,62 e 18,29 
# e localizados em Des Moines, uma cidade muito perto do centro de Seattle.
best_sqft_lot = df[(df['waterfront'] == True) & (df['view'] == 4) & (df['price'] <= 500000)].sort_values(by=['sqft_lot'], ascending=False)
best_sqft_lot.head()

id       date     price  bedrooms  bathrooms  sqft_living  \
18275  2781600195 2014-11-17  285000.0         1          1         1060   
11556  2013802030 2014-09-11  357000.0         3          2         2460   
8277    221029019 2015-04-28  400000.0         3          2         2090   
13525  2013802060 2014-09-27  500000.0         2          1         1760   
1949   1922039062 2015-04-20  480000.0         2          1         1008   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
18275     54846       1        True     4  ...        1060              0   
11556     53882       1        True     4  ...        2460              0   
8277      32718       2        True     4  ...        1550            540   
13525     27332       1        True     4  ...        1300            460   
1949      26487       1        True     4  ...        1008              0   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
18275      1935             0    98070  47.4716 -122.445           2258   
11556      1955             0    98198  47.3811 -122.325           2660   
8277       1919          1983    98070  47.3338 -122.511           1200   
13525      1951             0    98198  47.3799 -122.325           2590   
1949       1943          2002    98070  47.3853 -122.479           1132   

       sqft_lot15  price_per_area  
18275       31762        5.196368  
11556       32625        6.625589  
8277       192268       12.225686  
13525       16630       18.293575  
1949        24079       18.122098  

[5 rows x 22 columns]

In [114]:
# Analisando os imóveis nas mesmas condições, porém ordenando em ordem crescente de preço, encontrei imóveis baratos com áreas grandes para construção.
# A maioria localizado na ilha de Vashon, mas um me chamou atenção, o imóvel com id = 7631800110. Esse imóvel também foi vendido a um preço baixo, preço por area de 21,91 e possui
# muita possibilidade de crescimento por conta da localização (perto do centro de Seattle) e área do terreno.
best_price = df[(df['waterfront'] == True) & (df['view'] == 4) & (df['price'] <= 500000)].sort_values(by=['price'], ascending=True)
best_price.head()

id       date     price  bedrooms  bathrooms  sqft_living  \
18275  2781600195 2014-11-17  285000.0         1          1         1060   
11556  2013802030 2014-09-11  357000.0         3          2         2460   
264    2123039032 2014-10-27  369900.0         1          0          760   
18848  7631800110 2014-09-18  380000.0         3          2         1980   
8277    221029019 2015-04-28  400000.0         3          2         2090   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
18275     54846       1        True     4  ...        1060              0   
11556     53882       1        True     4  ...        2460              0   
264       10079       1        True     4  ...         760              0   
18848     17342       2        True     4  ...        1580            400   
8277      32718       2        True     4  ...        1550            540   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
18275      1935             0    98070  47.4716 -122.445           2258   
11556      1955             0    98198  47.3811 -122.325           2660   
264        1936             0    98070  47.4683 -122.438           1230   
18848      1984             0    98166  47.4551 -122.373           2060   
8277       1919          1983    98070  47.3338 -122.511           1200   

       sqft_lot15  price_per_area  
18275       31762        5.196368  
11556       32625        6.625589  
264         14267       36.700069  
18848       17313       21.912121  
8277       192268       12.225686  

[5 rows x 22 columns]

In [116]:
# Os últimos 2 imóveis que escolhi são: id = 6329000185 e id = 7936500221. Esses possuem um preço por area um pouco maior, porém satisfazem as mesmas condições dos outros.
city_seattle = df[
    (df['waterfront']==True) 
    & (df['view']==4) 
    & (
        (df['zipcode'] == 98117)  
        | (df['zipcode'] == 98118) 
        | (df['zipcode'] == 98119) 
        | (df['zipcode'] == 98121) 
        | (df['zipcode'] == 98122)
        | (df['zipcode'] == 98124)
        | (df['zipcode'] == 98125)
        | (df['zipcode'] == 98126)
        | (df['zipcode'] == 98129)
        | (df['zipcode'] == 98131)
        | (df['zipcode'] == 98131)
        | (df['zipcode'] == 98132)
        | (df['zipcode'] == 98133)
        | (df['zipcode'] == 98134)
        | (df['zipcode'] == 98136)
        | (df['zipcode'] == 98138)
        | (df['zipcode'] == 98144)
        | (df['zipcode'] == 98145)
        | (df['zipcode'] == 98146)
        | (df['zipcode'] == 98148)
        | (df['zipcode'] == 98151)
    )    
]
city_seattle.sort_values(by=['price'], ascending=True).head()

id       date     price  bedrooms  bathrooms  sqft_living  \
17474  6329000185 2015-03-29  540000.0         3          2         2600   
15371  7936500221 2015-01-14  658000.0         2          1         1010   
15031  7129303070 2014-08-20  735000.0         4          2         3040   
16418  6329000190 2014-07-29  750000.0         4          1         2520   
8224   7129303045 2015-04-17  949950.0         5          2         2340   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
17474     23361       1        True     4  ...        2150            450   
15371     14244       1        True     4  ...        1010              0   
15031      2415       2        True     4  ...        3040              0   
16418     21834       1        True     4  ...        1420           1100   
8224       1989       2        True     4  ...        2340              0   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
17474      1912             0    98146  47.4997 -122.379           1700   
15371      1926             0    98136  47.5476 -122.399           1820   
15031      1966             0    98118  47.5188 -122.256           2620   
16418      1960             0    98146  47.4996 -122.378           1700   
8224       1959             0    98118  47.5193 -122.257           2200   

       sqft_lot15  price_per_area  
17474       14700       23.115449  
15371       15792       46.194889  
15031        2433      304.347826  
16418        8100       34.350096  
8224         3230      477.601810  

[5 rows x 22 columns]

In [120]:
# Se olharmos os 10 imóveis mais caros do condado de King County, todos se localizam em Seattle ou muito perto de Seattle em um local mais urbanizado.
# Dessa forma tratei de escolher imóveis localizados nessas proximidades e com preços de venda baratos.
biggest_price = df[(df['waterfront'] == True) & (df['view'] == 4)].sort_values(by=['price'], ascending=False)
biggest_price.head(10)

id       date      price  bedrooms  bathrooms  sqft_living  \
1315   7558700030 2015-04-13  5300000.0         6          6         7390   
1164   1247600105 2014-10-20  5110800.0         5          5         8010   
8092   1924059029 2014-06-17  4668000.0         5          6         9640   
2626   7738500731 2014-08-15  4500000.0         5          5         6640   
2085   8106100105 2014-11-14  3850000.0         4          4         5770   
6508   4217402115 2015-04-21  3650000.0         6          4         5480   
15255  2425049063 2014-09-11  3640900.0         4          3         4830   
2864   4114601570 2014-11-18  3600000.0         3          3         5020   
21576  9253900271 2015-01-07  3567000.0         5          4         4850   
7989   9362000040 2014-06-23  3400000.0         3          4         5230   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
1315      24829       2        True     4  ...        5000           2390   
1164      45517       2        True     4  ...        5990           2020   
8092      13068       1        True     4  ...        4820           4820   
2626      40014       2        True     4  ...        6350            290   
2085      21300       2        True     4  ...        5770              0   
6508      19401       1        True     4  ...        3910           1570   
15255     22257       2        True     4  ...        4830              0   
2864      12431       2        True     4  ...        3420           1600   
21576     10584       2        True     4  ...        3540           1310   
7989      17826       2        True     4  ...        3740           1490   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
1315       1991             0    98040  47.5631 -122.210           4320   
1164       1999             0    98033  47.6767 -122.211           3430   
8092       1983          2009    98040  47.5570 -122.210           3270   
2626       2004             0    98155  47.7493 -122.280           3030   
2085       1980             0    98040  47.5850 -122.222           4620   
6508       1936             0    98105  47.6515 -122.277           3510   
15255      1990             0    98039  47.6409 -122.241           3820   
2864       1941          2002    98144  47.5925 -122.287           3680   
21576      2007             0    98008  47.5943 -122.110           3470   
7989       2005             0    98040  47.5348 -122.243           3670   

       sqft_lot15  price_per_area  
1315        24619      213.460067  
1164        26788      112.283323  
8092        10454      357.208448  
2626        23408      112.460639  
2085        22748      180.751174  
6508        15810      188.134632  
15255       25582      163.584490  
2864        12620      289.598584  
21576       18270      337.018141  
7989        17826      190.732638  

[10 rows x 22 columns]

In [139]:
# Considero esses os piores 5 imóveis pois se localizam na ilha de Vashon, uma ilha pouco povoada e longe do centro de Seattle, além disso são imóveis caríssimos,
# o que torna a possibilidade de lucro baixa.
piores_5_imoveis = df[(df['zipcode']==98070)].sort_values(by=['price'], ascending=False).head()

id       date      price  bedrooms  bathrooms  sqft_living  \
3934   1269200229 2014-07-23  1379900.0         3          3         3786   
49      822039084 2015-03-11  1350000.0         3          2         2753   
2964   2322029048 2014-11-19   999000.0         3          2         2830   
7250    722039049 2014-10-09   950000.0         4          3         3230   
13477   822039004 2014-06-13   849900.0         2          2         2280   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
3934      38038       1        True     4  ...        1934           1852   
49        65005       1        True     2  ...        2165            588   
2964     505166       1        True     3  ...        1830           1000   
7250     438213       2       False     0  ...        3230              0   
13477    641203       2       False     0  ...        2280              0   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
3934       1978          2006    98070  47.3907 -122.448           2850   
49         1953             0    98070  47.4041 -122.451           2680   
2964       1962             0    98070  47.3782 -122.514           2120   
7250       1999             0    98070  47.4141 -122.470           1600   
13477      1990             0    98070  47.4125 -122.455           2030   

       sqft_lot15  price_per_area  
3934        33361       36.276881  
49          72513       20.767633  
2964        21988        1.977568  
7250       144619        2.167896  
13477      224334        1.325477  

[5 rows x 22 columns]

In [127]:
# Esses são os 5 melhores imóveis que eu escolhi:
melhores_5_imoveis = df[(df['id']==6329000185) | (df['id']==7936500221) | (df['id']==7631800110) | (df['id']==2013802030) | (df['id']==2013802060)]
melhores_5_imoveis

id       date     price  bedrooms  bathrooms  sqft_living  \
11556  2013802030 2014-09-11  357000.0         3          2         2460   
13525  2013802060 2014-09-27  500000.0         2          1         1760   
15371  7936500221 2015-01-14  658000.0         2          1         1010   
17474  6329000185 2015-03-29  540000.0         3          2         2600   
18848  7631800110 2014-09-18  380000.0         3          2         1980   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
11556     53882       1        True     4  ...        2460              0   
13525     27332       1        True     4  ...        1300            460   
15371     14244       1        True     4  ...        1010              0   
17474     23361       1        True     4  ...        2150            450   
18848     17342       2        True     4  ...        1580            400   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
11556      1955             0    98198  47.3811 -122.325           2660   
13525      1951             0    98198  47.3799 -122.325           2590   
15371      1926             0    98136  47.5476 -122.399           1820   
17474      1912             0    98146  47.4997 -122.379           1700   
18848      1984             0    98166  47.4551 -122.373           2060   

       sqft_lot15  price_per_area  
11556       32625        6.625589  
13525       16630       18.293575  
15371       15792       46.194889  
17474       14700       23.115449  
18848       17313       21.912121  

[5 rows x 22 columns]

In [126]:
# Esse seria o gasto total: $2.435.000,00
gasto_total = sum(melhores_5_imoveis['price'])
gasto_total

2435000.0

In [132]:
# Essa é a média de valorização de cada imóvel: $1.727.691,00
media_valorizacao = biggest_price['price'].mean()
media_valorizacao

1727691.0518518519

In [133]:
# Esse é o retorno esperado: $8.638.455,00
retorno = media_valorizacao * 5
retorno

8638455.25925926

Piores 5 imóveis: 1269200229, 822039084, 2322029048, 722039049 e 822039004

Melhores 5 imóveis: 2013802030, 2013802060, 7936500221, 6329000185 e 7631800110.

*   Gasto = $2.435.000,00

*   Retorno = $8.638.455,00

*   Lucro = $6.203.455,00

# Rodrigo Oliveira do Monte Regato, RID: 37603
